packages/player/src/components/shortcut.tsx

```diff
@@ -0,0 +1,370 @@
+import * as React from 'react'
+import {VideoContext, VideoStateContext} from './player'
+import {useSelector} from '@xstate/react'
+import {State} from 'xstate'
+
+type ShortcutProps = {
+  clickable?: boolean
+  dblclickable?: boolean
+  shortcuts?: any[]
+}
+
+const selectHasStarted = (state: {context: VideoStateContext}) =>
+  state.context.hasStarted || false
+
+const selectIsActive = (state: {context: VideoStateContext}) =>
+  state.context.isActive || false
+
+const selectReadyState = (state: {context: VideoStateContext}) =>
+  state.context.readyState ?? -1
+
+const selectVolume = (state: {context: VideoStateContext}) =>
+  state.context.volume ?? 0.8
+
+const selectCurrentTime = (state: {context: VideoStateContext}) =>
+  state.context.video?.currentTime ?? 0
+
+const selectDuration = (state: {context: VideoStateContext}) =>
+  state.context.video?.duration ?? 0
+
+const selectIsPaused = (state: State<VideoStateContext>) =>
+  state.matches('ready.paused')
+
+const selectPlaybackRate = (state: State<VideoStateContext>) =>
+  state.context.playbackRate ?? 1.0
```

Comment:
xstate selectors that let us select "slices" of context as well as derived state.

https://xstate.js.org/docs/packages/xstate-react/#useselector-actor-selector-compare-getsnapshot